### PR TITLE
fix(logging): Add context to webhook logs

### PR DESCRIPTION
**Pull request by mcaulifn**
- Move metadata to JSON object for more structured logging
- Add additional detail to logs
- Use supportedEvents list to quickly exit if not supported

Closes #1369

### CODE REVIEW

File: modules/webhook/lambdas/webhook/src/webhook/handler.ts [hunk 1]
```diff
@@ -24,33 +25,60 @@ export async function handle(headers: IncomingHttpHeaders, body: string): Promis
...
+
+  if (payload[githubEvent].started_at) {
+    LogFields.fields.started_at = payload[githubEvent].started_at;
+  }
+
+  /*
+  The app subscribes to all `check_run` and `workflow_job` events.
+  If the event status is `completed`, log the data for workflow metrics.
+  */
+  if (payload[githubEvent].completed_at) {
```
Comment (npalm) [inline modules/webhook/lambdas/webhook/src/webhook/handler.ts:51]: Could you not replace the if statement by a single line like ```LogFields.fields.completed_at = payload[githubEvent]?.completed_at;```?

**Review by npalm — CHANGES_REQUESTED** [review]
No time to test yet

File: modules/webhook/lambdas/webhook/src/webhook/handler.ts [hunk 2]
```diff
@@ -24,33 +25,60 @@ export async function handle(headers: IncomingHttpHeaders, body: string): Promis
     return response;
   }
   const payload = JSON.parse(body);
-  logger.info(`Received Github event ${githubEvent} from ${payload.repository.full_name}`);
+  LogFields.fields.event = githubEvent;
+  LogFields.fields.repository = payload.repository.full_name;
+  LogFields.fields.action = payload.action;
+
+  if (!supportedEvents.includes(githubEvent)) {
+    logger.error(LogFields.fields, `Unsupported event type.`);
```
[the remark this answers is not in the document]
Comment (mcaulifn) [reply modules/webhook/lambdas/webhook/src/webhook/handler.ts]: Changed

**Review by npalm — COMMENTED** [review]
LGTM, will try to run a quick test tomorrow

**Review by npalm — DISMISSED** [review]

File: modules/webhook/lambdas/webhook/src/webhook/handler.ts [hunk 3]
```diff
@@ -24,33 +25,53 @@ export async function handle(headers: IncomingHttpHeaders, body: string): Promis
...
+  /*
+  The app subscribes to all `check_run` and `workflow_job` events.
+  If the event status is `completed`, log the data for workflow metrics.
+  */
+  LogFields.fields.completed_at = payload[githubEvent]?.completed_at;
+  LogFields.fields.conclusion = payload[githubEvent]?.conclusion;
 
   if (isRepoNotAllowed(payload.repository.full_name)) {
-    console.warn(`Received event from unauthorized repository ${payload.repository.full_name}`);
+    logger.error(LogFields.fields, `Received event from unauthorized repository ${payload.repository.full_name}`);
```
Comment (npalm) [inline modules/webhook/lambdas/webhook/src/webhook/handler.ts:52]: I think by a small change we can keep the log much cleaner and also readable. 

So would suggest to add an function to `logger.ts`
```TypeScript
  public static print(): string {
    return ' LogFields: ' + JSON.stringify(LogFields.fields);
  }
 ```
And next add the logfiels at the end of the line, for example 

```suggestion
    logger.error(`Received event from unauthorized repository ${payload.repository.full_name}. `, LogFields.fields);
```

**Review by npalm — CHANGES_REQUESTED** [review]
@mcaulifn change is in general fine to me. But would suggest to add the logfields to the end of the line. This to keep the logs also readable. See details on review comment

**Review by npalm — APPROVED** [review]
LGTM

**Pull request merged**